spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -245,6 +245,47 @@ public static Expression convert(Predicate predicate) {
     return null;
   }
 
+  private static <T> UnboundPredicate<T> handleEQPredicate(
+      Predicate predicate, BiFunction<UnboundTerm<T>, T, UnboundPredicate<T>> func) {
```

Comment:
I think this lambda makes it harder to read.  My vote is just we return a Pair (or better, a modeled class of value/columnName), and invoke the func directly after this method?

Probably the method name can be improved as well, as handle doesnt tell much, could be called something like columnAndValue, or something similar

Reply:
Inline the lambda in the `handleEQPredicate`.